src/credentials/attestation_verifier/JSONFileRevocationSet.cpp

```diff
@@ -0,0 +1,189 @@
+
+    if (revocationSetPath != nullptr)
+    {
+        LoadJSONRevocationSet(revocationSetPath, mRevocationSet);
+        VerifyOrReturn(revocationSetCount());
+    }
+
+    mIsInitialized = true;
+}
+
+void LoadJSONRevocationSet(const char * revocationSetPath, Json::Value & revocationSet)
+{
+    VerifyOrReturn(revocationSetPath != nullptr);
+
+    std::ifstream ifs(revocationSetPath);
+    Json::Reader reader;
+    reader.parse(ifs, revocationSet);
+}
+
+AttestationVerificationResult JSONFileRevocationSet::IsCertificateRevoked(bool isPaa, Crypto::AttestationCertVidPid vidPidUnderTest,
+                                                                          ByteSpan issuer, ByteSpan authorityKeyId,
+                                                                          ByteSpan serialNumber) const
+{
+    for (size_t revocation_set_idx = 0; revocation_set_idx < mRevocationSet.size(); ++revocation_set_idx)
+    {
+        Json::Value type = mRevocationSet[static_cast<int>(revocation_set_idx)]["type"];
+
+        // 1.
+        if (type.asString().compare("revocation_set") == 0)
+        {
+            Json::Value jsonCrlIssuerSubjectKeyId = mRevocationSet[static_cast<int>(revocation_set_idx)]["issuer_subject_key_id"];
+            Json::Value jsonCrlIssuerName         = mRevocationSet[static_cast<int>(revocation_set_idx)]["issuer_name"];
+            Json::Value jsonCrlRevokedSerialNumbers =
+                mRevocationSet[static_cast<int>(revocation_set_idx)]["revoked_serial_numbers"];
+
+            uint8_t crlIssuerSubjectKeyIdBuf[Crypto::kAuthorityKeyIdentifierLength] = { 0 };
+            ByteSpan crlIssuerSubjectKeyId(crlIssuerSubjectKeyIdBuf);
+
+            VerifyOrReturnError(Encoding::HexToBytes(jsonCrlIssuerSubjectKeyId.asString().c_str(),
+                                                     jsonCrlIssuerSubjectKeyId.asString().size(), crlIssuerSubjectKeyIdBuf,
+                                                     sizeof(crlIssuerSubjectKeyIdBuf)) == Crypto::kAuthorityKeyIdentifierLength,
+                                AttestationVerificationResult::kInvalidArgument);
+
+            size_t crlSignerCertificateMaxLength = BASE64_MAX_DECODED_LEN(jsonCrlIssuerName.asString().size());
+
+            Platform::ScopedMemoryBuffer<uint8_t> crlSignerCertificate;
+
+            ReturnErrorCodeIf(!crlSignerCertificate.Alloc(crlSignerCertificateMaxLength), AttestationVerificationResult::kNoMemory);
+
+            // 2.
+            size_t crlSignerCertificateLength =
+                Base64Decode(jsonCrlIssuerName.asString().c_str(), jsonCrlIssuerName.asString().size(), crlSignerCertificate.Get());
+            VerifyOrReturnError(crlSignerCertificateLength > 0 && crlSignerCertificateLength != UINT16_MAX,
+                                AttestationVerificationResult::kInternalError);
+
+            // 3.
+            if (isPaa)
+            {
+                Crypto::AttestationCertVidPid vid;
```

Comment:
Why is this `vid` if it's a vid/pid?

Reply:
1. Where are we checkign there is no PID in what we extract?
2. We are extracting two different things into the same AttestationCertVidPid.  Why are we doing that?

Reply:
1. Are you suggesting we should check that there's no PID available when evaluating a PAA?
2. Since we pass in DNAttrType::kCommonName as input, the first vidPid parameter is ignored. Only the second one is actually used.

Reply:
> Are you suggesting we should check that there's no PID available when evaluating a PAA?

I'm saying we should consider checking that, yes.  Is it valid to have a PID in there for a PAA?  If not, seems like we should fail out on garbage input.

> Since we pass in DNAttrType::kCommonName as input, the first vidPid parameter is ignored

The API documentation for ExtractVIDPIDFromAttributeString makes no such claims.  Should it be documenting that behavior, if it's part of the API contract?

Reply:
This file has been removed from the PR.